frontend/src/components/annotation/TaxonomicInput.tsx

```diff
@@ -0,0 +1,202 @@
+import { Autocomplete, capitalize, Grid, IconButton, TextField } from "@mui/material";
+import { FC, useState } from "react";
+import { useTranslation } from "react-i18next";
+import axios from 'axios';
+import HighlightOffIcon from '@mui/icons-material/HighlightOff';
+import { useAnnotationContext } from "../../contexts/annotationContext";
+import { Annotation } from "../../client/models/Annotation";
+
+
+const sep = " - ";
+
+interface TaxonomicInputProps {
+    rank: string;
+    observation: Annotation;
+};
+
+const TaxonomicInput: FC<TaxonomicInputProps> = (
+    props
+) => {
+    const { t } = useTranslation();
+
+    const { handleFormChange } = useAnnotationContext();
+    
+    const [load, setLoad] = useState<boolean>(false);
+    const [taxonList, setTaxonList] = useState<string[]>([]);
+    const [input, setInput] = useState<string>("");
+    
+    async function getData (search_name: string) {
+        let data:string[]=[];
+    
+        if (props.rank === "species") {
+            const speciesByLatinName = (await axios.get(`http://localhost:8889/taxapi/V1/taxons?RANG=ES&LB_NOM=${search_name}`)).data;
+            const speciesByVernName = (await axios.get(`http://localhost:8889/taxapi/V1/taxons?RANG=ES&NOM_VERN=${search_name}`)).data;
+            data = Array.from(new Set(speciesByLatinName.concat(speciesByVernName)));
+        };
+        if (props.rank === "genus") {
+            data = (await axios.get(`http://localhost:8889/taxapi/V1/taxons?RANG=GN&LB_NOM=${search_name}`)).data;
```

Comment:
And so I think it is better to isolate the API calls outside this component and so maybe "close" to the client directory but not into the client directory since it is autogenerated